### ISSUE TITLE
`notify` instead of `notify_change`?

### ISSUE THREAD

**Issue by rmorshea** [ipython/traitlets#382]
Should we `notify` or `notify_change`?

It seems like `notify_change` might be confusing as the types of notifications being used start to diversify. For instance, I can imagine passing a simple `Bunch(name='something', type='ping')` that doesn't signify a "change".

Of course, this wouldn't be backward compatible, but a simple regex pattern would suffice to account for this change downstream. If this change is deemed reasonable, one should not that `traitlets` still uses `_notify_change` internally, and that people may not have migrated to `notify_change` yet, so now would be the best time to make this change. Not later.

**Comment by minrk** [comment]
I think we are filling the namespace of objects too much. Every public method/attribute provided by HasTraits has a real cost to the quality of the APIs of classes that inherit from it, so we have to carefully consider any new method name. We should especially avoid taking up generic not-traitlets-specific names (like `notify`), because they mean that classes using traitlets cannot use those method names.

A better name would probably be `notify_trait`, but I think that was already taken and is now deprecated. It is better than either `notify` or `notify_change`, though.

Another question: Should the notify methods even be public? Shouldn't these only be called internally on change events?

**Comment by rmorshea** [comment]
The decision to make it public was made here: https://github.com/ipython/traitlets/pull/146

TLDR: ipywidgets overloads it, and I had been using it in matplotlib to mute in an outer context and then forcibly trigger from an inner context (something `hold_trait_notifications` didn't resolve).
